File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/dtos/CreateSubmissionRequest.java
package io.github.sanyavertolet.edukate.backend.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
public class CreateSubmissionRequest {
    @NotBlank
    private String problemId;
    @NotEmpty
    private List<@NotBlank String> fileNames;
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/dtos/CreateSubmissionRequest.kt
package io.github.sanyavertolet.edukate.backend.dtos

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty

data class CreateSubmissionRequest(
    @field:NotBlank val problemId: String,
    @field:NotEmpty val fileNames: List<@NotBlank String>,
)
